test/libyul/Parser.cpp

```diff
@@ -359,10 +379,26 @@ BOOST_AUTO_TEST_CASE(for_statement_continue_fail_post)
 	CHECK_ERROR_DIALECT(
 		"{ for {let i := 0} iszero(eq(i, 10)) {i := add(i, 1) continue} {} }",
 		SyntaxError,
-		"Keyword continue outside for-loop body is not allowed.",
+		"Keyword continue in for-loop init/post block is not allowed.",
 		dialect);
 }
 
+BOOST_AUTO_TEST_CASE(for_statement_continue_nested_init_in_body)
+{
+	auto dialect = EVMDialect::strictAssemblyForEVMObjects(EVMVersion::constantinople());
+	CHECK_ERROR_DIALECT(
+		"{ for {} 1 {} {let x for { continue } x {} {}} }",
+		SyntaxError,
+		"Keyword continue in for-loop init/post block is not allowed.",
+		dialect);
+}
+
+BOOST_AUTO_TEST_CASE(for_statement_continue_nested_body_in_init)
+{
+	auto dialect = EVMDialect::strictAssemblyForEVMObjects(EVMVersion::constantinople());
+	BOOST_CHECK(successParse("{ for {let x for {} x {} { continue }} 1 {} {} }", dialect));
+}
+
```

Comment:
For completeness, we can add the following test cases

```
BOOST_AUTO_TEST_CASE(for_statement_break_nested_body_in_init)
{
	auto dialect = EVMDialect::strictAssemblyForEVMObjects(EVMVersion::constantinople());
	BOOST_CHECK(successParse("{ for {let x for {} x {} { break }} 1 {} {} }", dialect));
}

BOOST_AUTO_TEST_CASE(for_statement_continue_nested_body_in_post)
{
	auto dialect = EVMDialect::strictAssemblyForEVMObjects(EVMVersion::constantinople());
	BOOST_CHECK(successParse("{ for {} 1 {let x for {} x {} { continue }} {} }", dialect));
}

BOOST_AUTO_TEST_CASE(for_statement_break_nested_body_in_post)
{
	auto dialect = EVMDialect::strictAssemblyForEVMObjects(EVMVersion::constantinople());
	BOOST_CHECK(successParse("{ for {} 1 {let x for {} x {} { break }} {} }", dialect));
}
```

Reply:
Added.